# NYC Urban Infrastructure & Service Analytics

## Key Findings & Business Insights

This notebook summarizes the key findings from the 2025 NYC 311 service request analysis.

The analysis integrates municipal service request data with geographic and socioeconomic information to examine:

* Service demand patterns
* Agency workload distribution
* Resolution time performance
* Borough and community board variation
* Complaint-type concentration
* Relationships between service request rates, socioeconomic characteristics, and resolution time

### Analysis Period

January 1, 2025 – December 31, 2025

### Analytical Approach

Findings in this notebook are based on validated datasets and previously completed SQL, geospatial, and socioeconomic analyses.

The analysis describes observed patterns and associations in the data. It does not infer causality from correlations or differences between geographic or socioeconomic groups.


In [3]:
from google.colab import drive

drive.mount('/content/drive')


Mounted at /content/drive


In [4]:
import os

powerbi_folder = "/content/drive/MyDrive/NYC_311_2025_BACKUP/POWERBI_DATA"

os.listdir(powerbi_folder)


['PowerBI_KPI_Summary.csv',
 'PowerBI_Monthly_Demand.csv',
 'PowerBI_Agency_Performance.csv',
 'PowerBI_Complaint_Performance.csv',
 'PowerBI_Borough_Performance.csv',
 'PowerBI_Community_Board_Performance.csv',
 'PowerBI_Socioeconomic_Analysis.csv']

In [5]:
import pandas as pd

kpi_summary = pd.read_csv(
    "/content/drive/MyDrive/NYC_311_2025_BACKUP/POWERBI_DATA/PowerBI_KPI_Summary.csv"
)

kpi_summary


,Metric,Value
0,Total Service Requests,3655040.00
1,Unique Service Requests,3655040.00
2,Agencies,15.00
3,Complaint Types,874.00
4,Community Boards,59.00
5,Boroughs,5.00
6,Valid Resolution Records,3525111.00
7,Average Resolution Hours,234.61
8,Resolved Within 24 Hours (%),62.18


In [6]:
monthly_demand = pd.read_csv(
    "/content/drive/MyDrive/NYC_311_2025_BACKUP/POWERBI_DATA/PowerBI_Monthly_Demand.csv"
)

agency_performance = pd.read_csv(
    "/content/drive/MyDrive/NYC_311_2025_BACKUP/POWERBI_DATA/PowerBI_Agency_Performance.csv"
)

complaint_performance = pd.read_csv(
    "/content/drive/MyDrive/NYC_311_2025_BACKUP/POWERBI_DATA/PowerBI_Complaint_Performance.csv"
)

borough_performance = pd.read_csv(
    "/content/drive/MyDrive/NYC_311_2025_BACKUP/POWERBI_DATA/PowerBI_Borough_Performance.csv"
)

community_board_performance = pd.read_csv(
    "/content/drive/MyDrive/NYC_311_2025_BACKUP/POWERBI_DATA/PowerBI_Community_Board_Performance.csv"
)

socioeconomic_analysis = pd.read_csv(
    "/content/drive/MyDrive/NYC_311_2025_BACKUP/POWERBI_DATA/PowerBI_Socioeconomic_Analysis.csv"
)

print("All analytical summary tables loaded successfully.")


All analytical summary tables loaded successfully.


In [7]:
# Executive Findings

print("TOTAL SERVICE REQUESTS:", int(kpi_summary.loc[
    kpi_summary["Metric"] == "Total Service Requests", "Value"
].iloc[0]))

print("AVERAGE RESOLUTION HOURS:", kpi_summary.loc[
    kpi_summary["Metric"] == "Average Resolution Hours", "Value"
].iloc[0])

print("RESOLVED WITHIN 24 HOURS (%):", kpi_summary.loc[
    kpi_summary["Metric"] == "Resolved Within 24 Hours (%)", "Value"
].iloc[0])

print("\nTOP 5 AGENCIES BY SERVICE REQUESTS")
display(
    agency_performance
    .sort_values("service_requests", ascending=False)
    .head(5)
)

print("\nTOP 5 COMPLAINT TYPES BY SERVICE REQUESTS")
display(
    complaint_performance
    .sort_values("service_requests", ascending=False)
    .head(5)
)

print("\nBOROUGH SERVICE REQUEST DISTRIBUTION")
display(
    borough_performance
    .sort_values("service_requests", ascending=False)
)


TOTAL SERVICE REQUESTS: 3655040
AVERAGE RESOLUTION HOURS: 234.61
RESOLVED WITHIN 24 HOURS (%): 62.18

TOP 5 AGENCIES BY SERVICE REQUESTS


,agency,service_requests,avg_resolution_hours,resolved_within_24h_pct
0,New York City Police Department,1717594,2.98,98.94
1,Department of Housing Preservation and Develop...,774535,367.99,13.56
2,Department of Sanitation,326780,89.75,38.67
3,Department of Transportation,212103,242.69,39.46
4,Department of Environmental Protection,197143,91.13,53.63



TOP 5 COMPLAINT TYPES BY SERVICE REQUESTS


,complaint_type,service_requests,avg_resolution_hours,resolved_within_24h_pct
0,Illegal Parking,564577,3.12,99.05
1,Noise - Residential,456599,3.31,98.50
2,HEAT/HOT WATER,315265,42.29,27.94
3,Blocked Driveway,169126,3.54,98.67
4,Noise - Street/Sidewalk,168021,1.81,99.77



BOROUGH SERVICE REQUEST DISTRIBUTION


,borough_clean,service_requests,avg_resolution_hours,resolved_within_24h_pct
0,BROOKLYN,1003412,201.90,61.36
1,QUEENS,813632,260.89,68.28
2,BRONX,781563,168.56,60.84
3,MANHATTAN,673197,309.63,57.64
4,UNKNOWN,258820,279.39,65.28
5,STATEN ISLAND,124416,260.94,56.23


## Executive Findings

### 1. Overall Service Demand

The analysis covers **3,655,040 NYC 311 service requests** recorded during 2025 across **15 agencies**, **874 complaint types**, **59 community boards**, and **5 boroughs**.

The overall average resolution time was **234.61 hours**, while **62.18%** of records with resolution information were resolved within 24 hours.

### 2. Agency Workload

Service demand was concentrated across a small number of agencies.

The **New York City Police Department** recorded **1,717,594 service requests**, followed by the **Department of Housing Preservation and Development** with **774,535** requests.

The Department of Sanitation, Department of Transportation, and Department of Environmental Protection followed with **326,780**, **212,103**, and **197,143** requests respectively.

Workload volume and resolution performance varied substantially across agencies, indicating that service demand and operational resolution patterns are not uniform across municipal functions.

### 3. Complaint-Type Concentration

The most frequent complaint categories were:

* **Illegal Parking:** 564,577 requests
* **Noise - Residential:** 456,599 requests
* **HEAT/HOT WATER:** 315,265 requests
* **Blocked Driveway:** 169,126 requests
* **Noise - Street/Sidewalk:** 168,021 requests

The high volume of parking and noise-related requests contrasts with the substantially longer average resolution time observed for HEAT/HOT WATER requests.

### 4. Borough-Level Variation

Brooklyn recorded the highest number of service requests at **1,003,412**, followed by Queens (**813,632**), the Bronx (**781,563**), and Manhattan (**673,197**).

Average resolution time also varied across boroughs. Manhattan recorded an average of **309.63 hours**, while the Bronx recorded **168.56 hours**.

These figures describe observed differences in the 2025 dataset and should not be interpreted as evidence that geography itself caused differences in resolution performance.

### Analytical Note

The findings above describe observed patterns in the dataset. Differences between agencies, complaint types, or boroughs may reflect differences in service type, workflow, reporting practices, case complexity, or other factors not directly measured in this analysis.


In [8]:
# Resolution Performance Findings

print("TOP 10 AGENCIES BY AVERAGE RESOLUTION HOURS")

display(
    agency_performance[
        ["agency", "service_requests", "avg_resolution_hours", "resolved_within_24h_pct"]
    ]
    .sort_values("avg_resolution_hours", ascending=False)
    .head(10)
)

print("\nTOP 10 COMPLAINT TYPES BY AVERAGE RESOLUTION HOURS")

display(
    complaint_performance[
        ["complaint_type", "service_requests", "avg_resolution_hours", "resolved_within_24h_pct"]
    ]
    .sort_values("avg_resolution_hours", ascending=False)
    .head(10)
)

print("\nTOP 10 COMMUNITY BOARDS BY AVERAGE RESOLUTION HOURS")

display(
    community_board_performance[
        ["community_board_clean", "service_requests", "avg_resolution_hours", "resolved_within_24h_pct"]
    ]
    .sort_values("avg_resolution_hours", ascending=False)
    .head(10)
)


TOP 10 AGENCIES BY AVERAGE RESOLUTION HOURS


,agency,service_requests,avg_resolution_hours,resolved_within_24h_pct
10,Economic Development Corporation,20554,7246.39,0.00
12,Department of Education,1737,2857.47,11.36
9,Taxi and Limousine Commission,36225,2320.26,8.67
6,Department of Buildings,103066,1315.21,25.40
5,Department of Parks and Recreation,110855,1241.87,22.78
13,Office of the Sheriff,1111,973.20,0.27
7,Department of Health and Mental Hygiene,83396,500.25,48.73
14,Office of Technology and Innovation,214,413.26,3.96
11,Department of Consumer and Worker Protection,18278,373.25,20.73
1,Department of Housing Preservation and Develop...,774535,367.99,13.56



TOP 10 COMPLAINT TYPES BY AVERAGE RESOLUTION HOURS


,complaint_type,service_requests,avg_resolution_hours,resolved_within_24h_pct
39,Noise - Helicopter,20049,7246.39,0.00
129,Taxi Report,790,5339.19,0.00
352,Water Maintenance,44,5282.36,9.09
685,FHV Licensee Complaint,3,5064.38,0.00
154,For Hire Vehicle Report,428,4966.45,0.00
327,Green Taxi Report,54,4746.97,0.00
439,Borough Office,23,4216.10,13.04
59,Taxi Complaint,9380,3393.02,0.06
668,DEP Highway Condition,4,3294.18,0.00
160,Green Taxi Complaint,377,3098.28,1.04



TOP 10 COMMUNITY BOARDS BY AVERAGE RESOLUTION HOURS


,community_board_clean,service_requests,avg_resolution_hours,resolved_within_24h_pct
20,10 QUEENS,56233,702.15,66.76
18,07 MANHATTAN,57383,675.63,51.67
51,06 MANHATTAN,33709,414.46,55.92
26,05 MANHATTAN,51983,331.10,65.41
57,11 QUEENS,29857,328.02,62.36
27,08 BROOKLYN,51712,327.46,52.11
47,08 MANHATTAN,40767,323.60,54.30
54,02 STATEN ISLAND,31738,310.60,59.45
37,13 QUEENS,48625,310.19,61.31
53,14 QUEENS,31790,298.00,57.46


## Resolution Performance Findings

### 1. Agency Resolution Patterns

Average resolution time varied considerably across agencies.

The **Economic Development Corporation** recorded the highest average resolution time at **7,246.39 hours**, followed by the **Department of Education** at **2,857.47 hours** and the **Taxi and Limousine Commission** at **2,320.26 hours**.

The **Department of Housing Preservation and Development**, which handled **774,535 requests**, recorded an average resolution time of **367.99 hours**, with **13.56%** of requests resolved within 24 hours.

These differences should be interpreted in the context of different agency responsibilities, service types, case complexity, and operational workflows.

### 2. Complaint Resolution Patterns

Several complaint categories recorded substantially longer average resolution times than the high-volume categories.

For example, **Noise - Helicopter** recorded an average resolution time of **7,246.39 hours** across **20,049 requests**. **Taxi Complaint** recorded **3,393.02 hours** across **9,380 requests**.

Some of the categories with the highest average resolution times had extremely small volumes, such as **FHV Licensee Complaint** with 3 requests and **DEP Highway Condition** with 4 requests.

Therefore, average resolution time for low-volume complaint categories should be treated as an observation rather than a strong operational conclusion.

### 3. Community Board Variation

Resolution performance also varied across community boards.

Among the boards with the highest average resolution times, **10 QUEENS** recorded **702.15 hours**, followed by **07 MANHATTAN** at **675.63 hours** and **06 MANHATTAN** at **414.46 hours**.

These differences represent observed variation in the 2025 data. Community board-level averages can also be influenced by the mix and complexity of service requests handled within each area.

### Analytical Note

Average resolution time is strongly affected by long-running cases and can therefore be highly skewed. For this reason, the dashboard also reports the percentage of requests resolved within 24 hours rather than relying on average resolution time alone.


In [9]:
# Socioeconomic Analysis Results

socioeconomic_correlations = pd.read_csv(
    "/content/drive/MyDrive/NYC_311_2025_BACKUP/NYC_311_2025_SOCIOECONOMIC_CORRELATIONS.csv"
)

poverty_quartiles = pd.read_csv(
    "/content/drive/MyDrive/NYC_311_2025_BACKUP/NYC_311_2025_POVERTY_QUARTILE_PROFILE.csv"
)

income_quartiles = pd.read_csv(
    "/content/drive/MyDrive/NYC_311_2025_BACKUP/NYC_311_2025_INCOME_QUARTILE_PROFILE.csv"
)

print("Correlation Results")
display(socioeconomic_correlations)

print("\nPoverty Quartile Profile")
display(poverty_quartiles)

print("\nIncome Quartile Profile")
display(income_quartiles)


Correlation Results


,relationship,n_tracts,pearson_r,spearman_rho
0,Request rate vs median household income,2189,-0.026006,-0.126343
1,Request rate vs poverty rate,2205,0.075789,0.170743
2,Request rate vs unemployment rate,2205,0.096683,0.124145
3,Median resolution vs median household income,2189,-0.000075,-0.131816
4,Median resolution vs poverty rate,2205,-0.011587,0.123162
5,Median resolution vs unemployment rate,2205,-0.012443,0.093274



Poverty Quartile Profile


,poverty_quartile,tracts,total_requests,median_requests_per_1000,median_resolution_days,median_p75_resolution_days,median_poverty_rate
0,Q1 - Lower Poverty,563,643793,288.716814,0.279213,2.894282,5.187747
1,Q2,563,728891,320.977011,0.294803,2.714063,10.944808
2,Q3,562,833237,349.750742,0.359696,2.884912,17.870747
3,Q4 - Higher Poverty,563,1061444,375.168086,0.541840,3.357361,32.314050



Income Quartile Profile


,income_quartile,tracts,total_requests,median_requests_per_1000,median_resolution_days,median_p75_resolution_days,median_household_income
0,Q1 - Lower Income,550,1071242,374.936089,0.610402,3.422876,45285.5
1,Q2,550,724196,337.563107,0.310845,2.583681,72291.0
2,Q3,548,691923,316.013670,0.274062,2.568924,93643.5
3,Q4 - Higher Income,550,726185,296.079556,0.325645,3.144771,134525.0


## Socioeconomic Findings

### 1. Request Rates and Poverty

The analysis examined **2,205–2,189 eligible census tracts**, depending on the socioeconomic variable.

The Spearman correlation between **service request rate and poverty rate was 0.171**, indicating a weak positive association.

The poverty-quartile analysis shows a similar pattern. Tracts in the lower-poverty quartile had a median request rate of **288.72 requests per 1,000 residents**, compared with **375.17 requests per 1,000 residents** in the higher-poverty quartile.

Median resolution time also increased across the poverty quartiles, from **0.279 days** in the lower-poverty quartile to **0.542 days** in the higher-poverty quartile.

### 2. Request Rates and Household Income

The Spearman correlation between **service request rate and median household income was −0.126**, indicating a weak negative association.

The income-quartile analysis shows that lower-income tracts had a median request rate of **374.94 requests per 1,000 residents**, compared with **296.08 requests per 1,000 residents** in higher-income tracts.

Median resolution time was **0.610 days** in the lower-income quartile and **0.326 days** in the higher-income quartile.

### 3. Resolution Time and Socioeconomic Characteristics

The relationship between socioeconomic characteristics and median resolution time was also weak.

The Spearman correlations were:

| Relationship                            | Spearman ρ |
| --------------------------------------- | ---------: |
| Median resolution vs. household income  |     −0.132 |
| Median resolution vs. poverty rate      |      0.123 |
| Median resolution vs. unemployment rate |      0.093 |

These values indicate relatively weak monotonic associations rather than strong relationships.

### 4. Interpretation

The socioeconomic analysis identifies measurable differences in service request rates and resolution patterns across tract-level socioeconomic groups.

However, these results should **not** be interpreted as evidence that poverty or income causes higher service demand or longer resolution times. The observed relationships may reflect multiple factors, including service composition, population characteristics, reporting behavior, geographic context, agency workflows, and other variables not included in this analysis.

The analysis therefore provides evidence of **association and variation**, not causation.


## Methodology & Limitations

### Data Sources

The project uses 2025 NYC 311 service request data as the primary municipal service dataset.

Additional geographic and socioeconomic analysis incorporates:

* NYC geographic boundary information
* U.S. Census American Community Survey (ACS) 2024 5-year data
* Census TIGER/Line geographic boundaries

### Data Processing

The 2025 NYC 311 dataset was processed through a structured data pipeline covering:

1. Data recovery and monthly consolidation
2. Data profiling
3. Data cleaning and validation
4. Geographic reconciliation
5. Census tract spatial analysis
6. Socioeconomic enrichment
7. SQL-based analytical analysis
8. Power BI dashboard preparation

The final validated dataset contains **3,655,040 service requests**.

### Geographic Analysis

Coordinate-valid service requests were spatially assigned to Census tracts using geographic boundary data.

The socioeconomic analysis included population, median household income, poverty rate, unemployment rate, housing units, service request rates, and resolution metrics.

Tracts with populations below 1,000 were excluded from the sensitivity-filtered socioeconomic analysis to reduce instability from very small population denominators.

### Resolution Metrics

Resolution time was calculated from service request creation and closure timestamps.

Both average resolution time and the percentage of requests resolved within 24 hours were examined because resolution times are strongly right-skewed and can be influenced by long-running cases.

### Statistical Analysis

Pearson and Spearman correlation measures were used to examine relationships between service request rates, resolution time, and socioeconomic characteristics.

Correlation results describe statistical association and **do not establish causation**.

### Limitations

Several limitations should be considered when interpreting the results:

* NYC 311 records represent reported service requests rather than all underlying conditions in the city.
* Resolution time may be influenced by agency workflows, case complexity, administrative processes, and reporting practices.
* Average resolution time can be heavily affected by unusually long-running cases.
* Socioeconomic variables are measured at the Census tract level and should not be interpreted as individual-level characteristics.
* Some socioeconomic variables contain incomplete coverage.
* Extremely small complaint categories can produce unstable average resolution times.
* Geographic analyses depend on the availability and quality of location information.
* Observed correlations should not be interpreted as causal relationships.

### Reproducibility

All major analytical stages were completed using documented notebooks, SQL analysis, validated summary datasets, and a Power BI dashboard.

The project is structured so that the analytical workflow can be reviewed and reproduced from the source data-processing stages through the final dashboard.


In [10]:
print("NYC Urban Infrastructure & Service Analytics")
print("08_Key_Findings.ipynb")
print("----------------------------------------")
print("Total Service Requests:", 3_655_040)
print("Agencies:", 15)
print("Complaint Types:", 874)
print("Community Boards:", 59)
print("Boroughs:", 5)
print("Average Resolution Hours:", 234.61)
print("Resolved Within 24 Hours (%):", 62.18)
print("----------------------------------------")
print("KEY FINDINGS NOTEBOOK COMPLETE")


NYC Urban Infrastructure & Service Analytics
08_Key_Findings.ipynb
----------------------------------------
Total Service Requests: 3655040
Agencies: 15
Complaint Types: 874
Community Boards: 59
Boroughs: 5
Average Resolution Hours: 234.61
Resolved Within 24 Hours (%): 62.18
----------------------------------------
KEY FINDINGS NOTEBOOK COMPLETE
